kubejs/server_scripts/base/featrues/chickenEggs.js

```diff
@@ -0,0 +1,31 @@
+// Change chicken egg generation from natural generation to being fed a seed
+
+// Set egg cooldown to max on spawn
+EntityEvents.spawned((event) => {
+  if (event.getEntity().getType() == "minecraft:chicken") {
+    event.getEntity().mergeNbt({ "EggLayTime": 2147483647 })
+  }
+});
+
+// Spawn an egg when a chicken is fed a seed
+ItemEvents.entityInteracted((event) => {
+  if (event.getTarget().getType() == "minecraft:chicken") {
+    let seeds = ["minecraft:wheat_seeds", "minecraft:melon_seeds", "minecraft:pumpkin_seeds", "minecraft:beetroot_seeds"];
```

Comment:
I think it's better if we used a tag for this, `'#forge:seeds'` includes those four already and if new seeds are added they will be added there